packages/salesforcedx-vscode-soql/src/queryDataView/queryDataFileService.ts

```diff
@@ -46,33 +47,40 @@ export class QueryDataFileService {
     }
   }
 
-  public save(): string {
+  public async save(): Promise<string> {
+    let queryDataSelectedPath = '';
     const fileContent = this.dataProvider.getFileContent(
       this.queryData.records
     );
-    const savedFileName = this.dataProvider.getFileName();
-    const queryDataFilePath = path.join(
-      this.getResultsDirectoryPath(),
-      savedFileName
+    const defaultFileName = this.dataProvider.getFileName();
+    /* queryDataDefaultFilePath will be used as the default options in the save dialog
+        fileName: The name of the soqlFile viewed in the builder
+        path: the same directory as the .soql file text doc.
+    note: directory must exist to show up in save dialog.
+    */
+    const queryDataDefaultFilePath = path.join(
+      path.parse(this.document.uri.path).dir,
+      defaultFileName
     );
 
-    this.createResultsDirectoryIfDoesNotExist();
-    // Save query results to disk
-    fs.writeFileSync(queryDataFilePath, fileContent);
-    this.showSaveSuccessMessage(savedFileName);
-    this.showFileInExplorer(queryDataFilePath);
-
-    return queryDataFilePath;
-  }
-
-  private getResultsDirectoryPath() {
-    return path.join(getRootWorkspacePath(), QUERY_RESULTS_DIR_PATH);
-  }
+    const fileInfo: vscode.Uri | undefined = await vscode.window.showSaveDialog(
+      {
+        defaultUri: vscode.Uri.file(queryDataDefaultFilePath)
+      }
+    );
 
-  private createResultsDirectoryIfDoesNotExist() {
-    fs.mkdirSync(this.getResultsDirectoryPath(), {
-      recursive: true
-    });
+    if (fileInfo) {
+      queryDataSelectedPath = fileInfo.path;
+      // Save query results to disk
+      const saveFilePath = path.join(fileInfo.path);
+      fs.writeFileSync(saveFilePath, fileContent);
+      // Only reveal saved file if its inside current workspace
```

Comment:
this logic should be in showFileInExplorer

Reply:
That makes sense, thanks.

Reply:
See the `fix: move logic to more relevant function` commit.